apis/addons/v1alpha1/addoninstances_types.go

```diff
@@ -0,0 +1,67 @@
+package v1alpha1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// AddonInstanceSpec defines the desired state of Addon operator.
+type AddonInstanceSpec struct {
+	// Name of Kubernetes Secret resource containing the credentials to Sendgrid
+	// +kubebuilder:default=10
+	HeartbeatUpdatePeriod int64 `json:"heartbeatUpdatePeriod,omitempty"`
+	// TODO: add more stuff here
+}
+
+type AddonInstancePhase string
+
+// Well-known Addon Phases for printing a Status in kubectl,
+// see deprecation notice in AddonStatus for details.
+const (
+	AddonInstancePhasePending     AddonInstancePhase = "Pending"
+	AddonInstancePhaseReady       AddonInstancePhase = "Ready"
+	AddonInstancePhaseTerminating AddonInstancePhase = "Terminating"
+	AddonInstancePhaseError       AddonInstancePhase = "Error"
+)
```

Comment:
I'm thinking if we should club this and [this](https://github.com/openshift/addon-operator/blob/main/apis/addons/v1alpha1/addons_types.go#L141) together and give them common names instead of having duplicates with different names

Reply:
I don't mind keeping them here for now, but FWIW, the `Phase` field will be deprecated soon once `kubectl` can print conditions. So I'm not sure why we'll continue refining phases.

https://github.com/openshift/addon-operator/blob/main/apis/addons/v1alpha1/addons_types.go#L131

Reply:
Ohh good point (missed it). Then, should we get rid of the `status.phase` field itself so as to stay compliant with non-deprecated pattern. 
cc: @thetechnick